--- /tmp/history.py	2024-03-24 13:38:37.247488754 +0100
+++ history.py	2024-03-24 13:42:26.455755394 +0100
@@ -6,6 +6,7 @@
 
 import os
 import time
+import datetime
 import contextlib
 import pathlib
 from typing import cast, Mapping, MutableSequence, Optional
@@ -134,10 +135,11 @@
 
     def __init__(self, database: sql.Database,
                  parent: Optional[QObject] = None) -> None:
-        super().__init__(database, "CompletionHistory", ['url', 'title', 'last_atime'],
+        super().__init__(database, "CompletionHistory", ['url', 'title', 'last_atime', 'human_time'],
                          constraints={'url': 'PRIMARY KEY',
                                       'title': 'NOT NULL',
-                                      'last_atime': 'NOT NULL'},
+                                      'last_atime': 'NOT NULL',
+                                      'human_time': 'NOT NULL'},
                          parent=parent)
         self.create_index('CompletionHistoryAtimeIndex', 'last_atime')
 
@@ -410,6 +412,7 @@
             return
 
         atime = int(atime) if (atime is not None) else int(time.time())
+        htime = datetime.datetime.now().strftime("%Y %B %d %H:%M:%S")
 
         with self._handle_sql_errors():
             self.insert({'url': self._format_url(url),
@@ -423,7 +426,8 @@
             self.completion.insert({
                 'url': self._format_completion_url(url),
                 'title': title,
-                'last_atime': atime
+                'last_atime': atime,
+                'human_time': htime
             }, replace=True)
 
     def _format_url(self, url):
